MusicTaste/MusicTaste/APIManagers/SpotifyAPIManager.m

```diff
@@ -150,26 +149,38 @@ - (NSDictionary *) convertSpotifyTracks:(NSArray *)tracks{
 }
 -(void) saveSpotifyData:(NSDictionary *) spotifyData{
     //Saves all Spotify Data such as there top genres,tracks,artist,album,and the artist images to the Parse Database
+    PFObject *music = [PFObject objectWithClassName:@"Music"];
     PFUser *current = [PFUser currentUser];
-    current[@"genres"] = [NSArray arrayWithArray:spotifyData[@"genres"]];
-    current[@"tracks"] = [NSArray arrayWithArray:spotifyData[@"tracks"]];
-    current[@"artists"] = [NSArray arrayWithArray:spotifyData[@"artists"]];
-    current[@"albums"] = [NSArray arrayWithArray:spotifyData[@"albums"]];
-    current[@"images"] = [NSArray arrayWithArray:spotifyData[@"images"]];
-    [PFUser.currentUser saveInBackground];
+    PFQuery *query = [PFQuery queryWithClassName:@"Music"];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *results, NSError *error) {
+        if (![[results valueForKey:@"userId"]containsObject:current.objectId]){
```

Comment:
This is where I make sure that there isn't a repeating user being added to the databse

Reply:
If they create a new user they wouldn't but this is great to think about. I would probably implement a way to refresh their preference after a certain time.